### PR TITLE
Refactoring ApiManager into custom element

### PR DESCRIPTION
**Pull request by yebrahim**
This is a continuation of the Module -> Custom Element refactor work, this time for the ApiManager class.

### CODE REVIEW

File: sources/web/datalab/polymer/components/datalab-files/datalab-files.ts [hunk 1]
```diff
@@ -120,6 +121,15 @@ class FilesElement extends Polymer.Element {
...
+    super();
+
+    // TODO: Should choose the FileManager instance dynamically here based on
+    // the data source specified for this datalab-files instance
+    this._fileManager = FileManagerFactory.getInstance();
+    this._apiManager = ApiManagerFactory.getInstance();
+  }
+
   /**
    * Called when when the element's local DOM is ready and initialized We use it
```
Comment (jimmc) [inline sources/web/datalab/polymer/components/datalab-files/datalab-files.ts]: typo: "when when"

File: sources/web/datalab/polymer/modules/file-manager/file-manager.ts [hunk 2]
```diff
@@ -42,9 +42,14 @@ interface Notebook {
...
-  FILE,
-  NOTEBOOK,
+  DIRECTORY = 'directory',
+  FILE = 'file',
+  NOTEBOOK = 'notebook',
+}
+
+enum DatalabFileStatus {
+  IDLE = '',
+  RUNNING = 'Running',
```
Comment (jimmc) [inline sources/web/datalab/polymer/modules/file-manager/file-manager.ts]: Are you directly using this string to display in the UI? Normally I would say that's a bad idea, but we are so far from having an internationalized UI that I don't know that it matters.

File: sources/web/datalab/polymer/components/datalab-files/datalab-files.ts (hunk 1 above)
Comment (yebrahim) [inline sources/web/datalab/polymer/components/datalab-files/datalab-files.ts]: Done.

File: sources/web/datalab/polymer/modules/file-manager/file-manager.ts (hunk 2 above)
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/file-manager/file-manager.ts]: Fixed by using a method for now, we'll think about moving that to a strings module.

**Pull request merged**